framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,242 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Poly2TriMeshGenerator.h"
+
+#include "CastUniquePointer.h"
+#include "MooseUtils.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/int_range.h"
+#include "libmesh/mesh_triangle_holes.h"
+#include "libmesh/parsed_function.h"
+#include "libmesh/poly2tri_triangulator.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseApp", Poly2TriMeshGenerator);
+
+InputParameters
+Poly2TriMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  MooseEnum algorithm("BINARY EXHAUSTIVE", "BINARY");
+
+  params.addRequiredParam<MeshGeneratorName>(
+      "boundary", "The MeshGenerator that defines the mesh outer boundary.");
+  params.addParam<unsigned int>(
+      "interpolate_boundary", 0, "How many more nodes to add in each outer boundary segment.");
```

Comment:
do you prefer interpolate over refine?
why not something explicit like num_nodes_per_segment ?

Reply:
I went with `add_nodes_per_boundary_segment` in the end; I figured `num` was misleading since this number isn't counting any part the nodes already at the beginning and end of each segment.

I think I've resolved all your issues now; would you mind double-checking and hitting "Resolve conversation" where ever I'm correct about that?  As a reviewer I sometimes find that an author's evaluation of "I've resolved this" doesn't match *my* evaluation, so I hate to be presumptuous/rude in the other direction.